src/lib/core/sequence.c

```diff
@@ -339,7 +339,16 @@ bt_sequence_get_machine_unchecked (const BtSequence * const self,
  * Return the length of the sequence minus any trailing rows that are totally
  * null/empty. Includes all available patterns, even those beyond the "end"
  * of the song.
- * 
+ *
+ * dbeswick: This function will be called whenever the sequence length is set,
+ * as part of https://github.com/Buzztrax/buzztrax/pull/109.
+ * It counts down from the max sequence length to try to find the last non-null
+ * pattern. This takes a long time if you happen to actually want a sequence that
+ * large, and tests were timing out, so I limited the max sequence size to 2**31.
+ *
+ * There are smarter ways to deal with this, but in the meantime I figure 2**31
+ * should be enough for anyone?
```

Comment:
+1 for that change. I think a better option could be to cache the max-length and update it for whenever there are sequence changes.